partiql-planner/src/main/kotlin/org/partiql/planner/internal/typer/DynamicTyper.kt

```diff
@@ -151,6 +163,34 @@ internal class DynamicTyper {
         }
     }
 
+    private fun computeDecimal(): DecimalType? {
+        val (precision, scale) = types.fold((0 to 0)) { acc, staticType ->
+            val decimalType = staticType as? DecimalType ?: return null
+            val constr = decimalType.precisionScaleConstraint as DecimalType.PrecisionScaleConstraint.Constrained
```

Comment:
Is there a need to check for `Unconstrained` in this situation? You do it further below in `computeString()`. This might throw a `ClassCastException` leading to unexpected runtime errors.

Especially since we know that CASE WHENs with multiple output types might lead to an unconstrained decimal. So, a nested CASE WHEN may very well end up in this situation.

Reply:
`computeDecimal` is only called when the `supertype` is a constrained `DECIMAL` (i.e. `PartiQLValueType.DECIMAL`) and not an arbitrary decimal (i.e. `PartiQLValueType.DECIMAL_ARBITRARY`). So a constrained decimal should never be in the list of `types` when we enter this block.

I can add the same logic from string constraints to prevent any accidental class cast exception in case there's a bug.

Reply:
Added in the v0.14.8 release PR